src/hooks/useAccessibilityNotification.ts

```diff
@@ -0,0 +1,122 @@
+import { uniqueId } from 'lodash';
+import { useCallback, useRef } from 'react';
+
+import styles from './accessibilityArea.module.scss';
+
+const notificationAreaId = 'accessibility-notifications-area';
+const activeAreasAttrKey = 'data-active-areas';
+
+const createNotificationAreaElement = (): HTMLElement => {
```

Comment:
Could this be done with a React component that would take of its own state instead of direct DOM manipulation?

Reply:
I re-implemented accessibility notifications with AccessibilityNotificationContext component. It uses contextAPI instead of direct DOM manipulation. I guess it's little bit more React-suitable approach